### DISCUSSION TITLE
JKBMS working via a TTL to USB in place of JKBMS-cable + 485 to USB converter

### DISCUSSION THREAD

**Discussion by githubfrance** [Louisvdw/dbus-serialbattery#524] [General]
I just only cut the cable and connected it directly to a TTL to USB and everything is working fine.
Any Idea how to connect the BMS directly to the Raspberry GPIOs ?

**Comment by mr-manuel** [comment]
If you don't know what you are doing it's better not to do it, else you can damage your Raspberry.

I don't know either, if the serial port (GPIO pin 8 for TX and 10 for RX) is enabled in Venus OS. It should appear as `/dev/serial0`. Would be very interesting, if this works out of the box.
The GPS port of the JKBMS provides the TTL signal. See https://github.com/Louisvdw/dbus-serialbattery/discussions/250#discussioncomment-4452179

Let us know your results!

**Comment by ramack** [comment]
I fear you could get into trouble if there is a ground shift, which in worst case works fine normally but as soon as the BMS cuts the B- sees a big offset. Using isolates connections would be good!

**Reply by githubfrance** [reply]
TTL to USB converter does the galvanic isolation

**Reply by mr-manuel** [reply]
Yes, but not if you connect directly to the GPIO pins

**Comment by githubfrance** [comment]
Hi Louis,

I cut the cable from the JKBMS 485 and kept only the side with the connector to the GPS. I connected the ground, TX and TX to the TTL to USB converter.
It has been running flawlessly for two days now.
![image](https://user-images.githubusercontent.com/18025874/232249176-2797737a-7a7e-4a3e-8eaf-d9f4ed56cc3b.png)

I also tested another TTL converter

![image (1)](https://user-images.githubusercontent.com/18025874/232249193-cd2e12f5-bfeb-4e26-ab55-70e61db3783d.png)
![image (2)](https://user-images.githubusercontent.com/18025874/232249267-6e4faa6b-f77f-4d5a-b5fd-522324dedd5e.png)

I will now look for a much smaller TTL converter and see if I can welded directly to the raspberry to keep it into the case.

I will see how to start the serial comm port on Venus OS.

Thank you for your work.
Best regards
Patrick

**Reply by mr-manuel** [reply]
We still cannot see the images. You only have to drag and drop them into the textfield.

**Reply by githubfrance** [reply]
at least !

**Reply by mr-manuel** [reply]
Now it works :D

**Reply by githubfrance** [reply]
I guess everything is obvious for you, but if I can answer questions, don't hesitate.

**Comment by githubfrance** [comment]
Today I did the same with a raspberry 3b, same positive result
